### DISCUSSION TITLE
the export button does not appear in the table

### DISCUSSION THREAD

**Discussion by yudisyudistira12** [pxlrbt/filament-excel#183] [Q&A]
`    
use pxlrbt\FilamentExcel\Actions\Tables\ExportBulkAction;

public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Select::make('user_id')
                    ->relationship('user', 'name')
                    ->required(),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('user.name')
                ->label('Name')
                ->searchable(),
                Tables\Columns\TextColumn::make('user_id')
                    ->label('Total Attendance')
                    ->summarize(Count::make('created_at')
                        ->label('Attendance'))
            ])
            ->bulkActions([
                ExportBulkAction::make()
            ])
            ->defaultGroup('user.name')
            ->groupsOnly()
            // Sorter By Name
            ->groups(['user.name'])
            ->filters([
                SelectFilter::make('month', 'Month')
                ->options([
                    'January' => 'January',
                    'February' => 'February',
                    'March' => 'March',
                    'April' => 'April',
                    'May' => 'May',
                    'June' => 'June',
                    'July' => 'July',
                    'August' => 'August',
                    'September' => 'September',
                    'October' => 'October',
                    'November' => 'November',
                    'December' => 'December',
                ])
                ->query(function(Builder $query, array $data){
                    $month = $data['value'] ?? 'January';
                    $year = now()->year;
            
                    switch ($month) {
                        case 'January':
                            return $query->whereBetween('created_at', ["$year-01-08", "$year-02-07"]);
                        case 'February':
                            return $query->whereBetween('created_at', ["$year-02-08", "$year-03-07"]);
                        case 'March':
                            return $query->whereBetween('created_at', ["$year-03-08", "$year-04-07"]);
                        case 'April':
                            return $query->whereBetween('created_at', ["$year-04-08", "$year-05-07"]);
                        case 'May':
                            return $query->whereBetween('created_at', ["$year-05-08", "$year-06-07"]);
                        case 'June':
                            return $query->whereBetween('created_at', ["$year-06-08", "$year-07-07"]);
                        case 'July':
                            return $query->whereBetween('created_at', ["$year-07-08", "$year-08-07"]);
                        case 'August':
                            return $query->whereBetween('created_at', ["$year-08-08", "$year-09-07"]);
                        case 'September':
                            return $query->whereBetween('created_at', ["$year-09-08", "$year-10-07"]);
                        case 'October':
                            return $query->whereBetween('created_at', ["$year-10-08", "$year-11-07"]);
                        case 'November':
                            return $query->whereBetween('created_at', ["$year-11-08", "$year-12-07"]);
                        case 'December':
                            return $query->whereBetween('created_at', ["$year-12-08", "$year-01-07"]);
                    }
                })
                ->default('January'),
            ]);
    }`

So, this my code for PresenceResource. i want to take action export data. but, the bulkaction button not show. how to fixed it ? i used laravel filament 3.

Thank

**Comment by pxlrbt** [comment]
Did you select any records?

**Comment by yudisyudistira12** [comment]
No, i don't select any record. but in various tutorials on YouTube, bulk action will appear if all data is selected first with checkboxes. Well, the problem is, my checkbox doesn't appear to the left of the name. how to handle it ?

**Reply by pxlrbt** [reply]
>  my checkbox doesn't appear to the left of the name. how to handle it ?

Did you overwrite BulkActions on the ListPage?

**Reply by yudisyudistira12** [reply]
Yes

**Reply by pxlrbt** [reply]
And that's empty? Then there's your answer.

**Comment by yudisyudistira12** [comment]
![Screenshot 2024-02-13 082331](https://github.com/pxlrbt/filament-excel/assets/59831982/69df41c9-50de-4153-a900-c15284a8ef23)

Maybe I explained it wrong. You can see, there is no checkbox to select the data that I have attached. The question is, how do I bring up the checkbox so that the data can be exported. How ?

**Reply by pxlrbt** [reply]
Please share your resources List Page code.

**Reply by yudisyudistira12** [reply]
`    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('user.name')
                ->label('Name')
                ->searchable(),
                Tables\Columns\TextColumn::make('user_id')
                    ->label('Total Attendance')
                    ->summarize(Count::make('created_at')
                        ->label('Attendance'))
            ])
            ->defaultGroup('user.name')
            ->groupsOnly()
            // Sorter By Name
            ->groups(['user.name'])
            ->filters([
                SelectFilter::make('month', 'Month')
                ->options([
                    'January' => 'January',
                    'February' => 'February',
                    'March' => 'March',
                    'April' => 'April',
                    'May' => 'May',
                    'June' => 'June',
                    'July' => 'July',
                    'August' => 'August',
                    'September' => 'September',
                    'October' => 'October',
                    'November' => 'November',
                    'December' => 'December',
                ])
                ->query(function(Builder $query, array $data){
                    $month = $data['value'] ?? 'January';
                    $year = now()->year;
            
                    switch ($month) {
                        case 'January':
                            return $query->whereBetween('created_at', ["$year-01-08", "$year-02-07"]);
                        case 'February':
                            return $query->whereBetween('created_at', ["$year-02-08", "$year-03-07"]);
                        case 'March':
                            return $query->whereBetween('created_at', ["$year-03-08", "$year-04-07"]);
                        case 'April':
                            return $query->whereBetween('created_at', ["$year-04-08", "$year-05-07"]);
                        case 'May':
                            return $query->whereBetween('created_at', ["$year-05-08", "$year-06-07"]);
                        case 'June':
                            return $query->whereBetween('created_at', ["$year-06-08", "$year-07-07"]);
                        case 'July':
                            return $query->whereBetween('created_at', ["$year-07-08", "$year-08-07"]);
                        case 'August':
                            return $query->whereBetween('created_at', ["$year-08-08", "$year-09-07"]);
                        case 'September':
                            return $query->whereBetween('created_at', ["$year-09-08", "$year-10-07"]);
                        case 'October':
                            return $query->whereBetween('created_at', ["$year-10-08", "$year-11-07"]);
                        case 'November':
                            return $query->whereBetween('created_at', ["$year-11-08", "$year-12-07"]);
                        case 'December':
                            return $query->whereBetween('created_at', ["$year-12-08", "$year-01-07"]);
                    }
                })
                ->default('January'),
            ])
            // ->headerActions([
            //     FilamentExportHeaderAction::make('export')
            // ])
            ->bulkActions([
                ExportBulkAction::make()
            ])
            ;
    }`
    
    
    Here's my code for table.

**Reply by pxlrbt** [reply]
Do you have duplicate code in Resource and your ListPage? Or is this your resource again? I was asking for the List Page. Btw. for multi line code blocks you need triple backticks not single

**Reply by yudisyudistira12** [reply]
No, i dont have duplicate code in resource. The list page has become one resource. 
![carbon (1)](https://github.com/pxlrbt/filament-excel/assets/59831982/d4349367-c72b-43a4-9235-31fed5a22941)

**Reply by yudisyudistira12** [reply]
And this my recently code for PresenceResource

**Reply by pxlrbt** [reply]
Yes, you already shared that 2 times. I am looking for the code of `ListPresences`

**Reply by yudisyudistira12** [reply]
Ohh.. ok. sorry. This my code for ListPresence
![carbon 3](https://github.com/pxlrbt/filament-excel/assets/59831982/2e509f80-4bb3-4e31-99cd-606114ec504e)

**Reply by pxlrbt** [reply]
Hm. Looks good to me. I am not sure what the issue is. Can you reproduce this on a fresh install with a minimal reproduction repo?

**Reply by yudisyudistira12** [reply]
I Can't.

**Reply by pxlrbt** [reply]
Hard to tell what's the issue then, if it's only in your current project.

**Reply by yudisyudistira12** [reply]
Ok, several days ago, i research whats my problem, and i got it the problem. Bulkactions can run if remove defaultGroup and groupsOnly. The data can be to export, but, in the display, i see a lot duplicate data. So, how if i don't delete defaultGroup and groupsOnly, the bulkactions can be run ?

**Reply by yudisyudistira12** [reply]
![Screenshot 2024-02-21 162031](https://github.com/pxlrbt/filament-excel/assets/59831982/143acd3a-7233-496d-bbd7-357b2fab62bb)

This pict if i still use groupsOnly and defaultGroup. 

![Screenshot 2024-02-21 162206](https://github.com/pxlrbt/filament-excel/assets/59831982/82257809-5fd0-4e8a-82b8-2475764183e3)


This pict if i remove groupsOnly and defaultGroup

**Reply by pxlrbt** [reply]
> So, how if i don't delete defaultGroup and groupsOnly, the bulkactions can be run ?

Sorry, I haven't used the grouping feature yet. Not sure whether it's supported with the export but maybe try the table header action instead of the bulk action.

**Comment by yudisyudistira12** [comment]
![carbon (1)](https://github.com/pxlrbt/filament-excel/assets/59831982/9f707a65-c644-48a2-b6e0-cb0844673f33)


This is my full code for PresenceResource

**Reply by pxlrbt** [reply]
Again: I am looking for the code for the ListPage. In your case this could be `ListPresences`
